src/PublishPort.jsx

```diff
@@ -7,13 +8,50 @@ import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput"
 import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
 import { OutlinedQuestionCircleIcon, TrashIcon } from '@patternfly/react-icons';
 import cockpit from 'cockpit';
+import ipaddr from "ipaddr.js";
+
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import * as utils from './util.js';
 
 const _ = cockpit.gettext;
 
-export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
+const MAX_PORT = 65535;
+
+export function validatePublishPort(value, key) {
+    switch (key) {
+    case "IP":
+        if (value && !ipaddr.isValid(value))
+            return _("Must be a valid IP address");
+        break;
+    case "hostPort": {
+        if (value) {
+            const hostPort = parseInt(value);
+            if (hostPort < 1 || hostPort > MAX_PORT)
+                return _("1 to 65535");
+        }
+
+        break;
+    }
+    case "containerPort": {
+        if (!value)
+            return _("Container port must not be empty");
+
+        const containerPort = parseInt(value);
+        if (containerPort < 1 || containerPort > MAX_PORT)
+            return _("1 to 65535");
+
+        break;
+    }
+    default:
+        console.error(`Unknown key "${key}"`);
```

Comment:
Please add a comment `// not-covered: unreachable assertion` to quiesce the coverage check. Also, please add "validatePublishPort" to the message, so that our future selves can identify where this actually happens.